xdsl/transforms/canonicalization_patterns/arith.py

```diff
@@ -16,3 +17,28 @@ def match_and_rewrite(self, op: arith.Addi, rewriter: PatternRewriter) -> None:
             and value.value.data == 0
         ):
             rewriter.replace_matched_op([], [op.rhs])
+
+
+class FoldConstConstOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(
+        self, op: arith.FloatingPointLikeBinaryOperation, rewriter: PatternRewriter, /
+    ):
+        if (
+            isinstance(op.lhs.owner, arith.Constant)
+            and isinstance(op.rhs.owner, arith.Constant)
+            and isa(l := op.lhs.owner.value, builtin.AnyFloatAttr)
+            and isa(r := op.rhs.owner.value, builtin.AnyFloatAttr)
+        ):
+            match type(op):
+                case arith.Addf:
+                    val = l.value.data + r.value.data
+                case arith.Subf:
+                    val = l.value.data - r.value.data
+                case arith.Mulf:
+                    val = l.value.data * r.value.data
+                case arith.Divf:
+                    val = l.value.data / r.value.data
```

Comment:
This will throw an exception if `r.value.data == 0` which is probably not desirable

Reply:
I just checked how upstream does it, and it appears they output `inf`

Reply:
That sounds reasonable to me. Python crashes by default so will have to manually check/catch